outline/shadowsocks/config.go

```diff
@@ -15,18 +15,25 @@
 package shadowsocks
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
 	"github.com/Jigsaw-Code/outline-ss-server/client"
 	"github.com/eycorsican/go-tun2socks/common/log"
 )
 
 // Config represents a shadowsocks server configuration.
-// Exported via gobind.
+// Exported via gobind.  Must match ShadowsocksSessionConfig interface in Typescript.
+//
+// TODO: Make this private, so platform code is not exposed to
+// Shadowsocks internals.
 type Config struct {
```

Comment:
Why are you concerned about breaking changes?

With https://github.com/Jigsaw-Code/outline-client/pull/1540 and https://github.com/Jigsaw-Code/outline-client/pull/1547 landed, we now have full IDE support for refactoring across these changes, so the update is extremely easy.  I wouldn't want to carry technical debt for the sake of avoiding those trivial changes.  The only case where we don't have that support (and where we've had a problem) is with the Electron child-process interface, but this PR retains the existing command-line arguments so there is no breaking change there.

Is the goal to improve our support for third-party consumers?  I don't think outline-go-tun2socks' API is ready to be stabilized for external consumption.

Reply:
nthlink is already using outline-go-tun2socks.
But it also affects us, as we saw with the prefix feature breakage. It prevents us to incrementally update platforms, since we pull one version for all platforms.

I don't think keeping the existing interface is a significant amount of technical debt. We can remove it later with a major version change when everything is settled (and nthlink moved to the new version).

Also, I feel strongly about naming the json structure `JSON`, since it needs specific consideration.

Reply:
OK, done

Reply:
Thanks. By the way, I looked for some projects using outline-go-tun2socks and found these:
- https://github.com/neputevshina/shadowban
- https://github.com/amnezia-vpn/desktop-client
- https://github.com/ASL-19/beepassvpn-client